dss/events/chunkedtask/_awstest.py

```diff
@@ -0,0 +1,56 @@
+import itertools
+import typing
+
+from ._awsimpl import AWSRuntime
+from .constants import TIME_OVERHEAD_FACTOR
+from .base import Task
+
+# The rest of this file is for unit tests.  The reason they are in this file is because they need to be deployed to the
+# lambda.  For production code, it may make sense to disable this, although it's pretty harmless.
+
+
+AWS_FAST_TEST_CLIENT_NAME = "fasttest"
+AWS_FAST_TEST_EST_TIME_MS = 100
+
+
+class AWSFastTestRuntime(AWSRuntime):
+    """
+    This is a modified variant of `AWSRuntime` that fakes less time being available than the system would otherwise
+    suggest.  The reason for this is to test the serialization and deserialization of state and the scheduling of work
+    without taking up the time fo a full lambda timeslice.  Furthermore, this reduces the risk that the test will
+    spuriously break at some future point in time if lambda lifetimes become longer.
+
+    This should only be used for the fast test.
+    """
+    def __init__(self, context, task_id: str) -> None:
+        super().__init__(context, AWS_FAST_TEST_CLIENT_NAME, task_id)
+        self.time_remaining_iterator = itertools.chain(
+            [int(AWS_FAST_TEST_EST_TIME_MS * TIME_OVERHEAD_FACTOR) + 1],
+            itertools.repeat(0)
+        )
```

Comment:
I know there are wonky ways to construct infinite generators, but those tend to rely on esoteric knowledge of python.

Do you have some other approach?